### PR TITLE
Wrote tests for dashboard functions: Caleb

### LINKED ISSUE #55: Create a test file for the database queries python file
As a developer, I want to have tests set up for my python script, so that I can be happy that my functions are working as intended and the script is suitable to be used in production.

### LINKED ISSUE #57: Create a test file for the dashboard graph/table creation python file
As a developer, I want to have tests set up for my python script, so that I can be happy that my functions are working as intended and the script is suitable to be used in production.

### LINKED ISSUE #59: Create a test file for the streamlit dashboard creation python file
As a developer, I want to have tests set up for my python script, so that I can be happy that my functions are working as intended and the script is suitable to be used in production.

### PR DESCRIPTION
**Pull request by CalebMathers**
This closes #59 
This closes #57 
This closes #55 

Wrote tests for the map query and creation functions in the aurora_map.py file. Removed some unnecessary files and fixed one test.

### CODE REVIEW

File: .github/workflows/py_test.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
     - name: "Install Python"
       uses: actions/setup-python@v5.1.1
       with:
-        python-version: '3.13'
+        python-version: '3.12'
```
Comment (olliejth) [inline .github/workflows/py_test.yml:23]: Good spot, change was ver necessary.

File: dashboard/test_aurora_map.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
+'''Contains tests for map generator functions.'''
```
Comment (olliejth) [inline dashboard/test_aurora_map.py:1]: Nice and concise.

File: .github/workflows/py_test.yml (hunk 1 above)
Comment (AlexOBarnes) [reply .github/workflows/py_test.yml:23]: Good spot

File: dashboard/test_aurora_map.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+        mock_get_connection.return_value.__enter__.return_value = mock_conn
+
+        result = get_average_cloud_data()
+        expected_result = {'County1': 60.5,
+                           'County2': 40.0}
+
+        assert result == expected_result
+
+
+class TestGetBodyVisibleRegions():
```
Comment (AlexOBarnes) [inline dashboard/test_aurora_map.py:139]: Good use of oop

**Review by AlexOBarnes — APPROVED** [review]

File: dashboard/test_aurora_map.py [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+
+        result = map_region_colours(['Test'])
+        expected_result = {'North East (England)': 'blue',
+                           'North West (England)': 'blue',
+                           'Northern Ireland': 'blue',
+                           'Scotland': 'blue',
+                           'Yorkshire and The Humber': 'blue',
+                           'Test': 'red'}
+
+        assert result == expected_result
```
Comment (olliejth) [inline dashboard/test_aurora_map.py:92]: Consistent naming.

File: dashboard/test_aurora_map.py [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+        mock_cursor.fetchall.return_value = [{'county_name': 'County1',
+                                              'avg_vis': 70.0},
+                                             {'county_name': 'County2',
+                                              'avg_vis': 55.0}]
+        mock_conn.cursor.return_value.__enter__.return_value = mock_cursor
+        mock_get_connection.return_value.__enter__.return_value = mock_conn
+
+        result = get_average_visibility_data()
+        expected_result = {'County1': 70.0,
+                           'County2': 55.0}
```
Comment (olliejth) [inline dashboard/test_aurora_map.py:233]: Neat and effective test object.

**Review by olliejth — APPROVED** [review]
If tests are passing then I see no issues.

**Pull request merged**